sound/soc/sof/pm.c

```diff
@@ -236,6 +236,15 @@ static int sof_resume(struct device *dev, bool runtime_resume)
 	if (!sdev->ops->resume || !sdev->ops->runtime_resume)
 		return 0;
 
+	/*
+	 * If for some reason, the cores were not powered off
+	 * during suspend, powering them up again will lead to
+	 * DSP panic. So check if core 0 is powered off to make
+	 * sure before proceeding further.
+	 */
+	if (snd_sof_dsp_core_is_enabled(sdev, BIT(0))
```

Comment:
This is starting to look like a hardware-dependent work-around in the framework. Can it be moved to the hardware or abstracted away?

Reply:
@plbossart sure. let me try to move it around. But I also feel like we are trying to hide the real issue with this hack. I am having second thoughts about it.